### PR TITLE
Wrapping up possible crashes 

### PR DESCRIPTION
**Pull request by phoinic**
Wrapped cases:

- "unwrap" usage - can cause panics
- assertions in parameters check - can cause panics
- direct calls of "panic!"

### CODE REVIEW

File: cctp_primitives/src/utils/mod.rs [hunk 1]
```diff
@@ -38,7 +46,9 @@ pub fn get_bt_merkle_root(bt_list: Option<&[BackwardTransfer]>) -> Result<FieldE
             let bt_fes = ByteAccumulator::init()
                 .update(bt)?
                 .get_field_elements()?;
-            assert_eq!(bt_fes.len(), 1);
+            if bt_fes.len() != 1 {
```
Comment (DanieleDiBenedetto) [inline cctp_primitives/src/utils/mod.rs:49]: Let's leave the assertion here

File: cctp_primitives/src/utils/mod.rs [hunk 2]
```diff
@@ -66,7 +76,10 @@ pub fn get_cert_data_hash(
         .update(btr_fee)?
         .update(ft_min_amount)?
         .get_field_elements()?;
-    assert_eq!(fees_field_elements.len(), 1);
+
+    if fees_field_elements.len() != 1 {
```
Comment (DanieleDiBenedetto) [inline cctp_primitives/src/utils/mod.rs:80]: Let's leave the assertion here

**Review by DanieleDiBenedetto — CHANGES_REQUESTED** [review]

File: cctp_primitives/src/utils/mod.rs (hunk 1 above)
Comment (phoinic) [reply cctp_primitives/src/utils/mod.rs:49]: Done

File: cctp_primitives/src/utils/mod.rs (hunk 2 above)
Comment (phoinic) [reply cctp_primitives/src/utils/mod.rs:80]: Done

**Review by DanieleDiBenedetto — APPROVED** [review]

**Comment by DanieleDiBenedetto** [comment]
@i-Alex if you can take a look at it too would be great. Some interfaces have slightly changed (mostly returning Err or Option because we removed some dangerous unwraps).
A parallel PR is opened on zendoo-sc-cryptolib too (https://github.com/HorizenOfficial/zendoo-sc-cryptolib/pull/61).
Please be aware of these changes when reporting them into the SDK (shouldn't be a huge deal).

**Review by i-Alex — APPROVED** [review]

**Pull request merged**